### ISSUE TITLE
Error: resource android:attr/fontVariationSettings not found.

### ISSUE THREAD

**Issue by nhohb07** [crosswalk-project/cordova-plugin-crosswalk-webview#207]
Hi guys,

The last week, I still built success project on Android, but now, I have an issue with run build on Android.

> `/Users/username/.gradle/caches/transforms-1/files-1.1/xwalk_core_library-22.52.561.4.aar/bd30a34e94f50c29b28a7ad58f86f45c/res/values/ui_java_dimens.xml
> Error:(14, 5) error: resource android:attr/fontVariationSettings not found.
> Error:(14, 5) error: resource android:attr/ttcIndex not found.
> /project/platforms/android/build/intermediates/incremental/mergeDebugResources/merged.dir/values/values.xml
> Error:(106) resource android:attr/fontVariationSettings not found.
> Error:(106) resource android:attr/ttcIndex not found.
> Error:failed linking references.
> Error:java.util.concurrent.ExecutionException: java.util.concurrent.ExecutionException: com.android.tools.aapt2.Aapt2Exception: AAPT2 error: check logs for details
> Error:java.util.concurrent.ExecutionException: com.android.tools.aapt2.Aapt2Exception: AAPT2 error: check logs for details
> Error:com.android.tools.aapt2.Aapt2Exception: AAPT2 error: check logs for details
> Error:Execution failed for task ':processDebugResources'.
> Failed to execute aapt`

Plugin list:

`<plugin name="cordova-plugin-device" spec="~1.1.1" />
>     <plugin name="cordova-plugin-whitelist" spec="~1.2.1" />
>     <plugin name="cordova-plugin-splashscreen" spec="~3.2.1" />
>     <plugin name="cordova-plugin-statusbar" spec="~2.1.0" />
>     <plugin name="ionic-plugin-keyboard" spec="~1.0.8" />
>     <plugin name="cordova-plugin-inappbrowser" spec="~1.2.0" />
>     <plugin name="cordova-plugin-google-analytics" spec="https://github.com/danwilson/google-analytics-plugin.git" />
>     <plugin name="cordova-plugin-crosswalk-webview" spec="^2.2.0">
>         <variable name="XWALK_VERSION" value="22+" />
>         <variable name="XWALK_LITEVERSION" value="xwalk_core_library_canary:17+" />
>         <variable name="XWALK_COMMANDLINE" value="--disable-pull-to-refresh-effect" />
>         <variable name="XWALK_MODE" value="embedded" />
>         <variable name="XWALK_MULTIPLEAPK" value="true" />
>     </plugin>
>     <engine name="ios" spec="^4.5.4" />
>     <engine name="android" spec="^6.4.0" />`

Please help me provide a solution on this issue.

Many thanks.

**Comment by nhohb07** [comment]
I try with suggest solution here: https://github.com/crosswalk-project/cordova-plugin-crosswalk-webview/issues/205#issuecomment-371669478

and now, I built success.

Thanks

**Comment by fongfuse** [comment]
try to change the compileSdkVersion to:  `compileSdkVersion 28`

**Comment by Madhuri98** [comment]
> try to change the compileSdkVersion to: `compileSdkVersion 28`

working thanks(y)

**Comment by ringatech** [comment]
updated  the code at platforms/android/app/build.gradle and worked like charm ;
dependencies {
    implementation fileTree(dir: 'libs', include: '*.jar')
    // SUB-PROJECT DEPENDENCIES START
    implementation(project(path: ":CordovaLib"))
    compile "com.android.support:support-v4:+"
    compile "com.android.support:support-v4:25.+"
    compile "com.android.support:appcompat-v7:25.+"
    // SUB-PROJECT DEPENDENCIES END
}

**Comment by MrDEV0** [comment]
It was fixed by updating this line `implementation "com.android.support:support-v4:+"`
to:
`implementation "com.android.support:support-v4:27.0.0"`

**Comment by nanox77** [comment]
I have `compileSdkVersion 28` , `minSdkVersion 21 `, `targetSdkVersion 28`. 
Also I have:
`configurations.all {
    resolutionStrategy {
        force 'com.android.support:support-v4:27.0.0'
    }
}`
and i have still the same error:

```
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':app_review:verifyReleaseResources'.
> java.util.concurrent.ExecutionException: com.android.builder.internal.aapt.v2.Aapt2Exception: Android resource linking failed
/Users/username/Documents/Flutter/big_stories/build/app_review/intermediates/res/merged/release/values/values.xml:249: error: resource android:attr/fontVariationSettings not found.
  /Users/username/Documents/Flutter/big_stories/build/app_review/intermediates/res/merged/release/values/values.xml:250: error: resource android:attr/ttcIndex not found.
  error: failed linking references.
```

flutter doctor -v

```[✓] Flutter (Channel stable, v1.2.1, on Mac OS X 10.14.2 18C54, locale en-AR)
    • Flutter version 1.2.1 at /Users/username/Documents/Flutter/flutter
    • Framework revision 8661d8aecd (3 weeks ago), 2019-02-14 19:19:53 -0800
    • Engine revision 3757390fa4
    • Dart version 2.1.2 (build 2.1.2-dev.0.0 0a7dcf17eb)

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/username/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)
    • All Android licenses accepted.

[!] iOS toolchain - develop for iOS devices (Xcode 10.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.1, Build version 10B61
    ✗ libimobiledevice and ideviceinstaller are not installed. To install with Brew, run:
        brew update
        brew install --HEAD usbmuxd
        brew link usbmuxd
        brew install --HEAD libimobiledevice
        brew install ideviceinstaller
    ✗ ios-deploy not installed. To install:
        brew install ios-deploy
    • CocoaPods version 1.5.3

[✓] Android Studio (version 3.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 33.3.1
    • Dart plugin version 182.5215
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)

[✓] IntelliJ IDEA Community Edition (version 2018.1.7)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 31.3.1
    • Dart plugin version 181.5656

[✓] VS Code (version 1.31.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 2.23.0

[✓] Connected device (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 9 (API 28) (emulator)

! Doctor found issues in 1 category.```

**Comment by vishakhavaidywan** [comment]
> try to change the compileSdkVersion to: `compileSdkVersion 28`
Mine is a cordova project. Can you tell me where to make this change? I cannot find this anywhere..

**Comment by pengcao1** [comment]
change the compileSdkVersion to: `compileSdkVersion 28`
resolved my issue

**Comment by marvinkosmider** [comment]
> change the compileSdkVersion to: `compileSdkVersion 28`
> resolved my issue

I have the same error and cannot find compileSdkVersion...

Edit:
Nevermind, i found it..

android/app/gradle.build

`//...`
`android {`
`  //...`
`  compileSdkVersion`

**Comment by anirudha-ani** [comment]
If anyone has stumbled upon this problem due to getting this error recently out of nowhere in your previously working fine ***react native projects***  - this is due to the latest BREAKING CHANGE in Google Play service and Firebase. Specially you will have problem with the packages which are dependent on -  

> com.google.android.gms:play-services-gcm

 Check this thread first -

https://github.com/facebook/react-native/issues/25293

And solution would mostly be like this -

https://github.com/facebook/react-native/issues/25293#issuecomment-503045776

**Comment by DarkKnight1992** [comment]
> If anyone has stumbled upon this problem due to getting this error recently out of nowhere in your previously working fine _**react native projects**_ - this is due to the latest BREAKING CHANGE in Google Play service and Firebase. Specially you will have problem with the packages which are dependent on -
> 
> > com.google.android.gms:play-services-gcm
> 
> Check this thread first -
> 
> [facebook/react-native#25293](https://github.com/facebook/react-native/issues/25293)
> 
> And solution would mostly be like this -
> 
> [facebook/react-native#25293 (comment)](https://github.com/facebook/react-native/issues/25293#issuecomment-503045776)

I almost gave up, since i have going to different thread for almost 4 days. This is really frustrating that this can break you app without any apparent reason.

check project.properties in the android project root folder for 
`com.google.android.gms:play-services-gcm` and `com.google.firebase:firebase-messaging` if it value is `com.google.android.gms:play-services-gcm:+` and `com.google.firebase:firebase-messaging:+` you would need downgrade play services possible to  `com.google.android.gms:play-services-gcm:16.1.0` and `com.google.firebase:firebase-messaging:18.0.0` should fix for most people if they problem occurred. You try different versions but making changes in the files itself to fully fix it you would need to identify the package using it and install is properly. Here is the link to the releases https://developers.google.com/android/guides/releases you can find the version which suits you.

**Comment by akildemir** [comment]
@pengcao1 @MarvinMakes I changed `compileSdkVersion cdvCompileSdkVersion` to `compileSdkVersion 28` and It stucks in after `License for package Android SDK Platform 28 accepted. Preparing "Install Android SDK Platform 28 (revision: 6)".` 

Also I tried to change  `compileSdkVersion cdvCompileSdkVersion` to `compileSdkVersion 28 cdvCompileSdkVersion`  and again it stucks after `License for package Android SDK Build-Tools 26.0.2 accepted. Preparing "Install Android SDK Build-Tools 26.0.2 (revision: 26.0.2)".` 

The versions were different but it doesn't install none of it. Did it install the version 28 in your case?
